### ISSUE TITLE
Filtering dataset

### ISSUE THREAD

**Issue by tentypwtk** [cymcsg/UltimateRecyclerView#398]
Hi,
 How can I add a feature of filtering through recyclerview dataset?
I want to iterate over the dataset, change a parameter, say "isVisible" for items that match, and hide other, notify the adapter about changes. I based my solution on ExpCustomAdapter, expandable RecyclerView, from your samples. Does it require extending of SmartItem code? I reckon there must be a simpler way to do that?

**Comment by jjhesk** [comment]
if you filter is an online query, you will need to have to do another query via http and have the items fetched from the result and update the adapter items. if you filter is doing offline items which is contained in the adapter, you will have to use the filter object to do the filtering works and making updates to the display results.

**Comment by tentypwtk** [comment]
Sorry for my misleading wording of the question. I'm comfortable with getting all the required data, my question was - what methods/objects of the UltimateRecyclerView to use to replace the currently visible data in the list. Should I try to remove items from mDataset (it's protected field) and add all the new items or there's a smarter way to do it?

**Comment by tentypwtk** [comment]
I'm now creating a new list with each filtering, creating a new adapter and assigning it to the recycler view (swapAdapter). If that's the way to go - fine. If that's super not effective - I'm open to suggestions how to get this done.
To make it clear - I'm filtering a list of topics with a search query from edittext.

**Comment by jjhesk** [comment]
yes. this is what i was doing for the filter adapter part at least that did not trigger any bugs related to this method.
